openwisp_monitoring/check/classes/ping.py

```diff
@@ -77,6 +79,14 @@ def check(self, store=True):
         timeout = self._get_param('timeout')
         ip = self._get_ip()
         if not ip:
+            device_monitored = DeviceMonitoring.objects.get(
+                device__pk=self.related_object.pk
+            )
+            if (
+                device_monitored.status in ['ok', 'problem']
+                and device_settings.CRITICAL_DEVICE_METRICS[0]['key'] == 'ping'

```

Comment:
if there are multiple metrics listed in this setting and ping is not the first one, this will fail.

The Metric model has a method to find out if a metric is critical, use that one next time.